connectors/salesforce/src/test/java/com/vmware/connectors/salesforce/SalesforceControllerTest.java

```diff
@@ -149,6 +148,47 @@ void testDiscovery() throws Exception {
         testConnectorDiscovery();
     }
 
+    @Test
+    void testRegex() throws Exception {
+        // https://blogs.msdn.microsoft.com/testing123/2009/02/06/email-address-test-cases/
+        List<String> expected = ImmutableList.of(
+                // VALID
+                "email001@domain.com",
+                "firstname.lastname002@domain.com",
+                "email003@subdomain.domain.com",
+                "firstname+lastname004@domain.com",
+                // TODO - the next 3 are valid apparently, but not worth fussing over
+//                "email005@123.123.123.123",
+//                "email006@[123.123.123.123]",
+//                "\"email007\"@domain.com",
+                "1234567890008@domain.com",
+                "email009@domain-one.com",
+                "_______010@domain.com",
+                "email011@domain.name",
+                "email012@domain.co.jp",
+                "firstname-lastname013@domain.com",
+
+                // INVALID? (in 2009 at least)
+//                "plainaddress014",
```

Comment:
Whether these commented lines are required? If not, please remove them.

Reply:
I want to keep them in so you can associate what is and isn't being matched in the input file easier.